desktop/src/main/java/org/destinationsol/desktop/SolDesktop.java

```diff
@@ -11,55 +26,96 @@
 import org.slf4j.LoggerFactory;
 import org.terasology.crashreporter.CrashReporter;
 
-import java.io.*;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
 import java.nio.charset.Charset;
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.text.SimpleDateFormat;
 import java.util.ArrayList;
-import java.util.Arrays;
 import java.util.Collections;
 import java.util.Date;
 import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * This class is the desktop entry point for the whole DestinationSol application. It handles the creation and launching
+ * of LwjglApplication from {@link SolApplication}.
+ */
+public final class SolDesktop {
 
-public class SolDesktop {
     private static Logger logger = LoggerFactory.getLogger(SolDesktop.class);
 
+    /**
+     * Specifies the commandline option to pass to the application for it to generate no crash reports. Coincidentally,
+     * it is the only commandline option detected by the application.
+     */
     private static final String NO_CRASH_REPORT = "-noCrashReport";
 
-    public static void main(String[] argv) {
-        List<String> commandLineArgs = Arrays.asList(argv);
+    /**
+     * This class is basically only a holder for the Java's {@code main(String[])} method, thus needs not to be
+     * instantiated.
+     */
+    private SolDesktop() {
+    }
 
+    public static void main(String[] argv) {
         LwjglApplicationConfiguration applicationConfig = new LwjglApplicationConfiguration();
-        boolean devBuild = java.nio.file.Files.exists(Paths.get("devBuild"));
-        if (devBuild) {
-            DebugOptions.DEV_ROOT_PATH = "engine/src/main/resources/"; // Lets the game run from source without a tweaked working directory
-            applicationConfig.vSyncEnabled = false; // Setting to false disables vertical sync
-            applicationConfig.foregroundFPS = 100; // Use 0 to disable foreground fps throttling
-            applicationConfig.backgroundFPS = 10; // Use 0 to disable background fps throttling
-        }
+        //TODO: Is checking for a presence of the file really the way we want to determine if it is a debug build?
```

Comment:
Probably not ... :D But it was the way it was written, until we have time to find a better way. Good TODO to keep around.

Reply:
I'm trying to add a bunch of trivial as well as not-so-trivial TODOs around  the place, so should it happen that we get a brand new contributor, he can easily start his learning of DS codebase with one of these.